substrate/frame/safe-mode/src/lib.rs

```diff
@@ -15,6 +15,48 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
+//! # Safe Mode
+//!
+//! Trigger for stopping all extrinsics outside of a specific whitelist.
+//!
+//! ## Pallet API
+//!
+//! See the [`pallet`] module for more information about the interfaces this pallet exposes,
+//! including its configuration trait, dispatchables, storage items, events, and errors.
+//!
+//! ## Overview
+//!
+//! Safe mode is entered via two paths (deposit or forced) until a set block number.
+//! The mode is exited when the block number is reached or a call to one of the exit extrinsics is
+//! made. A `WhitelistedCalls` configuration contains all calls that can be executed in while in
+//! safe mode.
+//!
+//! ### Primary Features
+//!
+//! - Entering safe mode can be permissionless with deposit, origin based, or some combination of
+//!   those.
+//! - Separate origin configuration for entering and exiting safe mode.
+//! - Safe mode must always be configured to exit at a specific block number.
+//! - Safe mode exit block number may be extended by additional calls.
+//!
+//! ### Example
```

Comment:
I think there could be more example, especially using `docify` to show how its used.  
You can probably just embed one of the basic tests for this.

Reply:
Tried setting this up. Does this look useful/correct to you?
![image](https://github.com/paritytech/polkadot-sdk/assets/1252199/10a4b102-83d3-4380-a2ef-9ce3e4eee4c1)